### PR TITLE
Delay start remote plugins until factory-plugin finish clone

### PR DESCRIPTION
**Pull request by evidolob**
### What does this PR do?
Delay start remote plugins until factory-plugin finish clone

### What issues does this PR fix or reference?
Related to eclipse/che#13427

### CODE REVIEW

**Comment by tolusha** [comment]
I am testing...

**Comment by tolusha** [comment]
It is hard to validate the issue, since it is needed to refresh the page to start LS :(

**Comment by dmytro-ndp** [comment]
Starting available Happy path tests: `che-theia-ci-test-k8s`

**Comment by dmytro-ndp** [comment]
che-theia-ci-test-k8s

File: extensions/eclipse-che-theia-plugin-remote/src/browser/remote-plugin-command.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import { injectable, inject } from 'inversify';
+import { CommandRegistry, CommandContribution, Command } from '@theia/core/lib/common';
+import { RemotePluginStarterService } from '../common/remote-plugin-protocol';
+
+const COMMAND: Command = {
+    id: '_remote_plugin_start_'
```
Comment (vitaliy-guliy) [inline extensions/eclipse-che-theia-plugin-remote/src/browser/remote-plugin-command.ts]: Could you please make the command ID consistent with other Theia command and name it something like `remote.plugin.start` ( or `remote.plugins.start` if it handles a set of plugins )

File: extensions/eclipse-che-theia-plugin-remote/src/browser/remote-plugin-command.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+@injectable()
+export class RemotePluginCommandContribution implements CommandContribution {
+
+    @inject(RemotePluginStarterService)
+    private starterService: RemotePluginStarterService;
+
+    registerCommands(commands: CommandRegistry): void {
+        commands.registerCommand(COMMAND, {
+            execute: () => {
+                this.starterService.loadRemotePlugins();
```
Comment (vitaliy-guliy) [inline extensions/eclipse-che-theia-plugin-remote/src/browser/remote-plugin-command.ts]: It's better to have
```
starterService.start()
```
It it's `starterService` or `pluginStarter` then better to name it `start`, if it's `pluginLoader` then `load`.

File: extensions/eclipse-che-theia-plugin-remote/src/common/remote-plugin-protocol.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+export const remotePluginServicePath = '/services/remotePlugin';
+
+export const RemotePluginStarterService = Symbol('RemotePluginStarterService');
+export interface RemotePluginStarterService {
```
Comment (vitaliy-guliy) [inline extensions/eclipse-che-theia-plugin-remote/src/common/remote-plugin-protocol.ts]: Is it better to have `RemotePluginLoader` of `RemotePluginInitializer`?

Comment (vitaliy-guliy) [reply extensions/eclipse-che-theia-plugin-remote/src/common/remote-plugin-protocol.ts]: It's just question of namings.

File: extensions/eclipse-che-theia-plugin-remote/src/browser/remote-plugin-command.ts (hunk 2 above)
Comment (vitaliy-guliy) [reply extensions/eclipse-che-theia-plugin-remote/src/browser/remote-plugin-command.ts]: Also the user can run this command using command palette. Could you prevent that by adding `isEnabled()` and `isVisible()` methods to command handler? Something like below
```
commands.registerCommand(COMMAND, {
  let initialized: boolean = false;
  execute: () => {
    initialized = true;
  },
  isEnabled() {
    return !initialized;
  },
  isVisible() {
    return !initialized;
  }
```

**Comment by l0rd** [comment]
@tsmaeder @tolusha please take the time today to review this PR. We should try to merge it today before @evidolob leave for PTO.

**Comment by tolusha** [comment]
@l0rd I can't validate the PR because java LS doesn't start

**Comment by tsmaeder** [comment]
> @l0rd I can't validate the PR because java LS doesn't start

@tolusha could you be a bit more precise? What are you trying to do and what doesn't work?

**Comment by tolusha** [comment]
@tsmaeder 
1. I've built images 
```
docker.io/eclipse/che-remote-plugin-runner-java8:next
docker.io/eclipse/che-theia:next
```
2. Put them into minishift
3. Set `CHE_WORKSPACE_SIDECAR_IMAGE__PULL__POLICY=IfNotPresent`
4. Started java workspace
5. Opened a java file
6. LS features didn't work

**Comment by evidolob** [comment]
@tsmaeder @l0rd During testing I figure out that https://github.com/eclipse/che/issues/13427 was not reproducing any more even without this PR. I think it can be related to activation events implementation in Theia, so now plugin(Java plugin with JDT-LS) start working only if user open java file. So maybe we do not need this PR?

**Comment by tsmaeder** [comment]
@evidolob Since it's a timing problem, it's hard to tell if it's "fixed". Only large scale testing will tell (folks who could reproduce before).
If there's a way to not need this PR, though, I would be relieved. It feels like kind of a hack.

**Comment by evidolob** [comment]
@tsmaeder I'm also prefer to not merge this PR, if the problem is gone. Before I faced this, but when we switch to Theia master with **che-theia:next** the problem is gone. 
We definitely should test more.

**Comment by l0rd** [comment]
I could reproduce https://github.com/eclipse/che/issues/13427 in the past (I have opened the issue after all) and was not able to reproduce the problem anymore (c.f. my comment https://github.com/eclipse/che/issues/13427#issuecomment-520275090). @rhopp and @eclipse/eclipse-che-qa could still reproduce it last week though.

Anyway @evidolob do you have any idea why @tolusha is not able to start the LS?

**Comment by sunix** [comment]
@tolusha Could you provide a complete devfile or at least let us know which projects you are cloning and which files you are opening ?

**Comment by tolusha** [comment]
@sunix 
That is the ordinary java maven stack

**Comment by dmytro-ndp** [comment]
crw-ci-test

**Comment by dmytro-ndp** [comment]
crw-ci-test

**Pull request closed without merging**